Classes/Domain/Model/Product.php

```diff
@@ -1075,19 +1075,19 @@ public function _getAllAttributeValues(): array // @codingStandardsIgnoreEnd
     }
 
     /**
-     * Find image by type.
+     * Find images by type.
      *
-     * @return Image|null
+     * @return Image[]|null
      */
-    protected function findImageByType(int $type): ?Image
+    protected function findImagesByType(int $type): ?array
```

Comment:
This method should also return an empty array only, not null.

Also, renamed `$imagesArray` to `$images`. You don't need to put the data type in the variable name.